### PR TITLE
ci: add codecov

### LINKED ISSUE #513: ci: set up codecov

### PR DESCRIPTION
**Pull request by tmc**
Fixes #513

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -4,6 +4,7 @@ go:
 - 1.7.x
```
Comment (achew22) [inline .travis.yml]: I think we should drop 1.7.x right now and drop 1.8.x when 1.10 is released

Comment (tmc) [reply .travis.yml]: +1

**Comment by tmc** [comment]
Hm, oddly hitting an infinite recursion issue when trying to run coverage.

**Comment by achew22** [comment]
I think we should remove the 1.10 beta from the test matrix. I'm sure we aren't the only people hitting that and we don't have release notes on how to fix this yet

**Comment by tmc** [comment]
I wanted/needed it for coverage output consolidation. (-coverpkg=all)

**Pull request merged**